fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/DimensionIdFilter.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.resolver;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.util.StreamUtils;
+import com.yahoo.bard.webservice.web.ApiFilter;
+import com.yahoo.bard.webservice.web.FilterOperation;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * An implementation of {@link DataSourceFilter} which uses dimensionApiFilters to match partition data sources.
+ * The matching algorithm requires that for each dimension specified, the query filters must match one of the
+ * corresponding values.
+ */
+public class DimensionIdFilter implements DataSourceFilter {
+
+    private final Map<Dimension, ApiFilter> dimensionKeySelectFilters;
+
+    /**
+     * Constructor.
+     *
+     *
+     * @param dimensionMappingValues  The map of dimension to sets of dimension values which map this table.
+     */
+    public DimensionIdFilter(Map<Dimension, Set<String>> dimensionMappingValues) {
+        dimensionKeySelectFilters = dimensionMappingValues.entrySet().stream()
+                .map(entry -> new AbstractMap.SimpleEntry<>(
+                        entry.getKey(),
+                        new ApiFilter(entry.getKey(), entry.getKey().getKey(), FilterOperation.in, entry.getValue())
```

Comment:
It's likely not worth it here, since this is a fairly simple constructor, and the call is likely very cheap. The added complexity isn't worth it, imho.